packages/emotion/src/index.js

```diff
@@ -170,9 +173,27 @@ function createStyles(strings, ...interpolations) {
       styles += strings[i + 1]
     }
   })
+
+  return styles
+}
+
+function identityParser(selector, styles) {
   return styles
 }
 
+function buildAndInsertStyles(selector, styles, parser) {
+  if (process.env.NODE_ENV === 'production') {
+    parser(selector, styles)
+  } else {
+    const index = styles.indexOf('/*# sourceMappingURL')
+    if (index > -1) {
+      currentSourceMap = styles.substr(index)
+    }
+    parser(selector, styles)
+    currentSourceMap = ''
+  }
+}
+
```

Comment:
Maybe instead of putting the sourcemap inside of the styles we could make an object the first argument to createStyles if there is a sourceMap and do something like this.
```jsx
let sourceMap = ''

function createStyles(strings, ...interpolations) {
  if (process.env.NODE_ENV !== 'production') {
    if (strings !== undefined && strings.sourceMap !== undefined) {
      sourceMap = strings
      return createStyles(...interpolations)
    }
  }
  let stringMode = true
  let styles = ''
  if (
    (strings !== undefined && strings.raw === undefined) ||
    strings === undefined
  ) {
    stringMode = false
    styles = handleInterpolation(strings, false)
  } else {
    styles = strings[0]
  }
  interpolations.forEach((interpolation, i) => {
    styles += handleInterpolation(interpolation, isLastCharDot(styles))
    if (stringMode === true && strings[i + 1] !== undefined) {
      styles += strings[i + 1]
    }
  })
  return styles
}
```

Reply:
This is close to what I did.